### ISSUE TITLE
QtCreator Compile Issues

### ISSUE THREAD

**Issue by vk6mo** [pa2eon/QT6SSTV#4]
Hi Eric
Building works fine from the prompt.
Next step - get building working from within QtCreator.
So I install QtCreator (v10.0.1)
plus install doxygen (apt-get install doxygen).
Then try to build from QtCreator and I get 
In file included from ../src/scope/scopeplot.h:25,
                 from ../src/scope/scopeview.h:36,
                 from ../src/sstv/sstvrx.cpp:12:
./ui_plotform.h:21:10: fatal error: qwt_plot.h: No such file or directory
   21 | #include "qwt_plot.h"
      |          ^~~~~~~~~~~~
Obviously this relates to library libqwt but which **version** should be installed???
And where do I get it from??
The only one I can see that is available is for qt5!! (ie libqwt-qt5-dev)??

**Comment by pa2eon** [comment]
Hi Dave,
I have checked the versions on my system:
Installed is : libqwt-headers libqwt-qt5-dev libqwt-qt5-6 - installed version is ended at 6.1.4-2

**Comment by vk6mo** [comment]
I have downloaded qwt v6.2.0 from https://qwt.sourceforge.io/ and then compiled and installed.
Which sort of works except everything is located where QtCreator can't find it...
Any suggestions as to best way to proceed?

Link below might help
https://qwt.sourceforge.io/qwtinstall.html

THis is where QWT installs by default.
QWT_INSTALL_PREFIX    = /usr/local/qwt-$$QWT_VERSION
QWT_INSTALL_DOCS      = $${QWT_INSTALL_PREFIX}/doc
QWT_INSTALL_HEADERS   = $${QWT_INSTALL_PREFIX}/include
QWT_INSTALL_LIBS      = $${QWT_INSTALL_PREFIX}/lib

**Comment by vk6mo** [comment]
> Hi Dave,
> I have checked the versions on my system:
> Installed is : libqwt-headers libqwt-qt5-dev libqwt-qt5-6 - installed version is ended at 6.1.4-2

Can you build from Creator?

**Comment by pa2eon** [comment]
Yep no problem .. BUT use my advises in the start page of this github .. to mention the PATH of the QT creator.
Keep the setup as clean as possible .. The version 6.4.2 will give you also the latest qmake (Vs. 6.4.2 ) when you want to make the software by CLI.

**Comment by vk6mo** [comment]
Following your setup I had conflicts between qmakes etc
export export PATH=/opt/Qt/6.x.x/gcc_64/bin/:$PATH => 6.x.x is for example 6.5.0  - this causes problems even for command line build and then you can't do the ln command as qmake already exists.

ln -s /opt/Qt/6.x.x/gcc_64/bin/qmake /usr/bin/qmake   --- this is good for me

What I have put in the wiki works for me, but that is only for command line build

According to qwt page, we need to use qwt 6.2 for Qt6 support
https://qwt.sourceforge.io/qwtchangelog.html

**Comment by pa2eon** [comment]
>>> Following your setup I had conflicts between qmakes etc
export export PATH=/opt/Qt/6.x.x/gcc_64/bin/:$PATH => 6.x.x is for example 6.5.0 - this causes problems even for command line build and then you can't do the ln command as qmake already exists. <<<

That is correct .. the version 6.5.0 is NOT correct to use .. I have changed it in front the read.me

**Comment by vk6mo** [comment]
> > > > Following your setup I had conflicts between qmakes etc
> > > > export export PATH=/opt/Qt/6.x.x/gcc_64/bin/:$PATH => 6.x.x is for example 6.5.0 - this causes problems even for command line build and then you can't do the ln command as qmake already exists. <<<
> 
> That is correct .. the version 6.5.0 is NOT correct to use .. I have changed it in front the read.me

I didn't use 6.5.0 - I was just using your example text

I did 
export PATH=/opt/Qt/6.2.4/gcc_64/bin/:$PATH - works OK and hides old stuff as /opt/Qt/6.2.4/gcc_64/bin/ is first.
ln -s /opt/Qt/6.x.x/gcc_64/bin/qmake /usr/bin/qmake - BAD!! because /usr/bin/qmake already exists 

On front page, do you mean Qt v6.4.2 or Qt v6.2.4??

**Comment by vk6mo** [comment]
For me.....
![image](https://github.com/pa2eon/QT6SSTV/assets/132327266/1fd1064f-5b9a-4368-885e-4e20b4573150)

**Comment by pa2eon** [comment]
You have to remove the logic link first .. then the last command will be fine ..Op 7 jun. 2023 15:59 schreef Dave ***@***.***>:
For me.....

**Comment by vk6mo** [comment]
If you need to remove the link, then something is not right....
I think this is **not** a Qt issue, this is a **qwt** issue. Creator cannot find qwt.
What directory is your qwt installed in?
What version of Creator are you using?
What output do you get from "qmake -query"??
How does your creator find the qwt?

Below is from the qsstv.pro file
!macx: INCLUDEPATH += /usr/include/qwt /usr/include/qt5/qwt
!macx: LIBS += -lqwt-qt5
It is still referencing qt5!!!!

I think what should be happening is qwt is added as a **feature**
Then the
CONFIG += qwt
picks up the settings that are necessary for QtCreator to find the libraries etc.
(see https://qwt.sourceforge.io/qwtinstall.html)

So the user should type (this is persistent so only needs to be done once)

- qmake -set QMAKEFEATURES=${QWT_ROOT}/features

(for me it was /usr/local/qwt-6.2.0/features)

then we comment out/remove the following two lines

!macx: INCLUDEPATH += /usr/include/qwt /usr/include/qt5/qwt
!macx: LIBS += -lqwt-qt5

then the build finds both qwt headers and qwt library.

What do you think?
If you are happy with this I will put it all together into the wiki

**Comment by vk6mo** [comment]
Hi Eric
I've put the QtCreator instructions in p4 of the Wiki - see what you think.
When I get a chance I will roll back my test machine and follow the instructions to see everything works following the exact instructions.
Dave

**Comment by vk6mo** [comment]
Hi Eric
Just to let you know I have rolled back my linux PC and followed the instructions as per page 4 of the wiki.
I have also commented out
!macx: INCLUDEPATH += /usr/include/qwt /usr/include/qt5/qwt
!macx: LIBS += -lqwt-qt5
(we will need to push this change back into the repository).

Once done, the code compiles and runs from QtCreator.

Now we can start looking at the code..! :)

I will close this issue in a couple of days as its resolved.
